node/src/components/consensus/era_supervisor.rs

```diff
@@ -1040,18 +1042,26 @@ impl EraSupervisor {
                     .cloned()
                     .collect();
                 let random_bit = rng.gen();
-                effect_builder
-                    .request_appendable_block(block_context.timestamp())
-                    .map(move |appendable_block| {
-                        Arc::new(appendable_block.into_block_payload(accusations, random_bit))
-                    })
-                    .event(move |block_payload| {
-                        Event::NewBlockPayload(NewBlockPayload {
-                            era_id,
-                            block_payload,
-                            block_context,
-                        })
+                let delay_by = self.proposal_delay_failpoint.fire(rng).cloned();
```

Comment:
I think I'd swap this with the line below, just to keep failpoint logic as close as possible, but that's an unimportant nitpick. :)

Have you considered changing the future (no pun intended) instead, to keep the failpoint logic grouped tigher together? That may be worse though since the future will probably need to be boxed in that case. 

The `.map` might be moved into `async` block as well, to avoid having to repeat that, or the `Arc`ification be moved into the closure found in `.event`.